include/gridtools/common/hugepage_alloc.hpp

```diff
@@ -21,16 +21,8 @@ namespace gridtools {
      * reduce cache set conflicts.
      */
     inline void *hugepage_alloc(std::size_t size) {
-#ifdef __cray__
-// see issue https://github.com/GridTools/gridtools/issues/1557
-#warning "hugepage_alloc on Cray might be suboptimal"
-#endif
-        static std::atomic<std::size_t> s_offset(64);
-        auto offset = s_offset.load(std::memory_order_relaxed);
-        auto next_offset = offset;
-        while (!s_offset.compare_exchange_weak(
-            next_offset, 2 * next_offset <= 4096 ? 2 * next_offset : 64, std::memory_order_relaxed)) {
-        }
+        static std::atomic<std::size_t> s_counter(0);
+        auto offset = 64 << (s_counter++ %7);
```

Comment:
Hmm while looking at this code again, I was wondering if it actually makes sense to do it that way. It comes directly from Intel, so maybe yes, but on the other hand it is to avoid L1 cache conflicts and we explicitly restrict us to only 6 out of 64 available sets which does not make a lot of sense. I will open an issue about it, I also observed that the allocation does not use huge pages on some systems were they would be available, so I might file a PR at some point.